spec/google/gax/api_callable_spec.rb

```diff
@@ -71,10 +76,29 @@ module Gax
     end
   end
 
+  describe 'create_api_call with block options' do
+    it 'calls with block' do
+      adder = 0
+      settings = CallSettings.new
+
+      op = double('op', request: 3)
+      allow(op).to receive(:execute) { 2 + op.request + adder }
+
+      func = proc do |request, _deadline: nil, **_kwargs|
+        expect(request).to eq(3)
+        op
+      end
+
+      my_callable = Google::Gax.create_api_call(func, settings)
+      expect(my_callable.call(3)).to eq(5)
+      expect(my_callable.call(3) { adder = 5 }).to eq(5)
```

Comment:
This test is confusing, since you have two different `adder` variables in two different scopes, and the `adder` in the block in this line just gets discarded.